zeppelin-web/src/app/visualization/builtins/visualization-barchart.js

```diff
@@ -115,39 +128,16 @@ export default class BarchartVisualization extends Nvd3ChartVisualization {
       configObj.changeXLabel('default')
     }
 
+    if (typeof (configObj.rotate.degree) === 'undefined' || configObj.rotate.degree === '') {
+      configObj.rotate.degree = '-45'
+      self.emitConfig(configObj)
+    }
+
     return {
-      template: `<div>
-          xAxis :
-      </div>
-
-      <div class='btn-group'>
-        <button type="button"
-              class="xLabel btn btn-default btn-sm"
-              ng-class="{'active' : this.config.isXLabelStatus('default')}"
-              ng-click="save('default')" >
-            Default
-        </button>
-
-        <button type="button"
-              class="btn btn-default btn-sm"
-              ng-class="{'active' : this.config.isXLabelStatus('rotate')}"
-              ng-click="save('rotate')" >
-            Rotate
-        </button>
-
-        <button type="button"
-              class="btn btn-default btn-sm"
-              ng-class="{'active' : this.config.isXLabelStatus('hide')}"
-              ng-click="save('hide')" >
-            Hide
-        </button>
-      </div>`,
+      template: `<ng-include src="'app/visualization/builtins/visualization-dispayXAxis.html'">
```

Comment:
[`template` field understand URL](https://github.com/apache/zeppelin/blob/master/zeppelin-web/src/app/visualization/visualization.js#L156). Could you try it without 'ng-include'?